activerecord/test/cases/associations/belongs_to_associations_test.rb

```diff
@@ -1199,6 +1203,47 @@ def test_belongs_to_invalid_dependent_option_raises_exception
     assert_equal "The :dependent option must be one of [:destroy, :delete, :destroy_async], but is :nullify", error.message
   end
 
+  class DestroyByParentBook < ActiveRecord::Base
```

Comment:
I think these tests are good, but the naming is confusing. In this case, it is the child (book) that is being destroyed, so I think it would be better if the classes such as "DestroyByParentBook/Author" were renamed something like "DestroyByChildBook/Author" or even better "DestroyViaBelongsToBook/Author"